src/consensus/aft/impl/view_change_tracker.h

```diff
@@ -103,22 +103,55 @@ namespace aft
     kv::Consensus::SeqNo write_view_change_confirmation_append_entry(
       kv::Consensus::View view)
     {
-      auto it = view_changes.find(view);
-      if (it == view_changes.end())
+      ccf::ViewChangeConfirmation nv =
+        create_view_change_confirmation_msg(view);
+      return store->write_view_change_confirmation(nv);
+    }
+
+    std::vector<uint8_t> get_serialized_view_change_confirmation(
+      kv::Consensus::View view)
+    {
+      ccf::ViewChangeConfirmation nv =
+        create_view_change_confirmation_msg(view);
+      nlohmann::json j;
+      to_json(j, nv);
+      std::string s = j.dump();
+
+      std::vector<uint8_t> ret;
+      ret.resize(s.size() + 1);
```

Comment:
Why +1?

Reply:
I cleaned this up to use a vector ctor but the + 1 is needed so that we copy the string terminator.